.gitignore

```diff
@@ -78,5 +78,10 @@ security/cmd/node_agent/na/cert_file
 security/cmd/node_agent/na/pkey
 # Test artifacts
 tests/e2e/local/minikube/docker-machine-driver-hyperkit
+/istio_bin/
```

Comment:
I'm not sure I want us to go back to 'dirty' work directory. 

Any reason to create temp files in the source directory instead of a work directory as we have today ?

Reply:
Reduced the number of changed. Now it's pretty much just fixing the code to honor the TOP environment variable and getting rid of unused/redundant stuff in env package.

Note that makefiles still don't support TOP, but Go-only change will at lest make some use cases work (like direct go test invocations).